### PR TITLE
Add ZeroG Ladder floor type

### PR DESCRIPTION
**Pull request by abackwood**
Adds a ZeroG Ladder type floor, much like the rungs and handholds seen on the outside of real-life space stations for astronauts to move around.

Eventually, this tile is meant to have the capability of letting characters move around as normal on the outside, whereas normal floors don't provide friction or are in another way less useful (details dependent on later implementation of "moving on the outside" of course).

Ladders can only be built on the outside. They don't allow furniture to be built on them. Characters can walk on them and one tile around them just like normal floors.

A button was added to the build UI to build ladders. For now this button is only present in a separate test scene. I'm not sure how to merge this into the main scene without causing major conflicts. The FloorMenu was changed to include the ladder build button.

zerogladderscreenshot

### CODE REVIEW

**Comment by Tranberry** [comment]
In a future PR the 'ladder' might be made like the wall sprite I guess?

File: Assets/Scripts/Controllers/BuildModeController.cs [hunk 1]
```diff
@@ -214,11 +223,27 @@ public void DoBuild(Tile t)
         }
         else
         {
-            Logger.LogError("UNIMPLMENTED BUILD MODE");
+            Logger.LogError("UNIMPLEMENTED BUILD MODE");
         }
 
     }
 
+    // Checks whether the given floor type is allowed to be built on the tile
```
Comment (Randshot) [inline Assets/Scripts/Controllers/BuildModeController.cs]: Comments should end with a period.

File: Assets/Scripts/Models/Tile.cs [hunk 2]
```diff
@@ -98,7 +99,7 @@ public float movementCost
 
                 foreach (Tile n in ns) // Loop through all the horizontal/vertical neighbours of the empty tile.
                 {
-                    canMove = canMove || (n != null && n.Type == TileType.Floor); // If the neighbour is a floor tile, set canMove to true.
+                    canMove = canMove || (n != null && (n.Type == TileType.Floor || n.Type == TileType.Ladder)); // If the neighbour is a floor tile, set canMove to true.
```
Comment (Randshot) [inline Assets/Scripts/Models/Tile.cs]: Comments should be on their own line.

File: Assets/Scripts/UI/InGameUI/FloorMenu.cs [hunk 3]
```diff
@@ -17,6 +17,12 @@ void Start()
             {
                 bmc.SetMode_BuildFloor();
             });
+        if(ladderBuild != null) {
```
Comment (Randshot) [inline Assets/Scripts/UI/InGameUI/FloorMenu.cs]: Curly braces should always be on a new line.

**Comment by gunthergun** [comment]
Love the ladders <3

**Comment by Mizipzor** [comment]
Really like this. :)

**Comment by gunthergun** [comment]
Why did you commit changes to the scene ?

**Comment by abackwood** [comment]
It was an accident. I changed something in the scene in the first commit and then spent a few commits reverting it. The scene is in sync with the master branch but I think a timestamp got changed in the meta file

**Comment by Mizipzor** [comment]
Getting the scene changes out from this pull request will require some intermediate git-fu. Hopefully its accepted as in.

**Comment by Randshot** [comment]
@abackwood Maybe you could also add the german translations :)
`build_ladder=Leiter bauen
 tile_Ladder=Leiter
 tile_Ladder_desc=Neue Fortbewegung-Einrichtung. Effektivität wird stark durch schwache Arme reduziert`

Something like this.

**Comment by WardBenjamin** [comment]
Looks awesome, thanks!

**Comment by WardBenjamin** [comment]
@abackwood I'll add this to the main scene before I go to bed tonight, thanks!

**Pull request merged**